lib/teiserver/microblog/libs/post_lib.ex

```diff
@@ -194,11 +210,16 @@ defmodule Teiserver.Microblog.PostLib do
 
   """
   def delete_post(%Post{} = post) do
-    query = "DELETE FROM microblog_post_tags WHERE post_id = $1;"
-    {:ok, _} = Ecto.Adapters.SQL.query(Repo, query, [post.id])
+    Repo.transaction(fn ->
+      query = "DELETE FROM microblog_post_tags WHERE post_id = $1;"
+      {:ok, _} = Ecto.Adapters.SQL.query(Repo, query, [post.id])
+
+      query = "DELETE FROM microblog_poll_responses WHERE post_id = $1;"
+      {:ok, _} = Ecto.Adapters.SQL.query(Repo, query, [post.id])
 
-    Repo.delete(post)
-    |> broadcast_delete_post
+      Repo.delete(post)
+      |> broadcast_delete_post
+    end)
```

Comment:
The return of the transaction is `{:ok, stuff} | {:error, tx_error}`, where stuff is whatever is returned from the tx block. So when that works all fine, you get `{:ok, {:ok, post}}`. You need to pattern match on that to unwrap the extra `:ok, ...` tuple.
This is the source of the test failure at `test/teiserver/microblog/microblog_test.exs:133`